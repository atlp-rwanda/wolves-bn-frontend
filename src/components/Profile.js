import React, { Component } from 'react';

class Profile extends Component {
  render() {
    return (
      <div >
        <h2> Hello </h2>
      </div>
    );
  }
}

export default Profile;
