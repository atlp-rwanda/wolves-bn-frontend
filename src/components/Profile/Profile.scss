.container{
    width: 60%;
    margin: auto;
    height: auto;
    padding: 0;
    display: flex;
    border-radius: 10px;
    align-items: center;
    justify-content: space-around;
    //background-color: #fff;
    box-shadow: 0 10px 20px rgba(#000, 0.2);

  }
  .profile{
    background-image: url(../../assets/images/bg.png);
    background-size: cover;
    background-position: center;
    display: flex;
    padding: 50px;
    align-content: space-between;
  } 

  .img-fluid{
    width: 100%;
    height: 100%;
    padding: 0 10%;
  }
  .img-container{
    width: 50%;
  }
  .mmmm{
    width: 100%;
    margin: 0 0.5rem;
    h3{
       color: #000000;
       margin-bottom: 10px;
       font-family: poppins;
       
    }
    h5{
      color: red;
      padding: 10px auto;
      font-family: poppins;

    }
    h4{
      color:green;
      margin-bottom: 10px;
      margin-top: -10px;
      font-family: poppins;

    }
  }
    form{
    min-width: 100%;
    box-shadow: 2px 2px 5px #2C166F;
    padding:20px;
    margin: 0;
    
  }
//   form{
//     width: 50%;
//     padding: 50px;
//     box-shadow: 2px 2px 5px #2C166F;
// }
  .content{
    width: 90vw;
    background-color: white;
    display: flex;
    justify-content: space-around;
    padding: 4rem 1rem;
    // box-shadow: 0 0 5px 5px rgba(0,0,0, .05);
  }
  .form-control{
    display: block;
    width: 100%;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    border-color: #2196F3 !important;
    border-style: solid !important;
    border-width: 0 0 1px 0 !important;
    padding: 0px !important;
    color: #495057;
    height: auto;
    border-radius: 0;
    background-color: white;
    background-clip: padding-box;
  }
  
  .form-control:focus{
    color: #495057;
    outline: none;
    border-color: white;
    background-color: white;
    box-shadow: none; 
  }
  .form-control input{
    width: 70%;
    padding: auto;
  }
 
.infoCard {
    position: relative;
    width: 50%;
    margin: 120px auto;
    background-color: #fff;
    box-shadow: 0 10px 20px rgba(#000, 0.2);
    padding: 60px 30px;
    text-align: center;
    border-radius: 10px;
    top: 60px;

    button {
        position: absolute;
        top: 10px;
        right: 20px;
        outline: none;
        Link{
          color:blue;
          background: none;
          
        }
    }

}




.userIcon {
    position: absolute;
    top: -50px;
    left: 50%;
    transform: translateX(-50%);
    border-radius: 50%;

    img {
        width: 100px;
        height: 100px;
        border-radius: 50%;
    }
}

.btn {
  background-color: #0029FA;
  color: white;
  font-size: 18px;
  margin: 10px auto;
  margin-left: 20px;
  width: fit-content;
  border: none;
  border-radius: 5px;
  button:hover{
  background-color: #5C73F2;
  cursor: pointer;
  }
}


.btn--primary {
    background-color: #0029FA;
}

.userName {
    margin-bottom: 40px;
    font-family: poppins;

    span {
        display: inline-block;
        font-weight: 500;
        font-size: 24px;
        font-family: poppins;
        margin-bottom: 10px;
        margin-left: 10px;
    }

    p {
        font-weight: 100;
        
    }
}

.moreInfo {
    span {
        display: inline-block;
        margin-bottom: 10px;
        font-family: poppins;
        margin-left: 10px;
    }

    .role {
        color: #000;
    }
}

.error {
    color: red;
}