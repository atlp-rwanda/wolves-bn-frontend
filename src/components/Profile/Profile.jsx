import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { viewProfile } from '../../redux/actions/profile/profile';
import './Profile.scss';

export class Profile extends Component {
  componentDidMount() {
    this.props.fetchProfile();
  }

  render() {
    const {
      firstName, lastName, address, email, phone, role, currency, department, gender, profileimage,
    } = this.props.user;

    return (
      <div className='profile'>
        <div className="infoCard">
          <div className="userIcon">
              <img src={profileimage} alt="user Icon"/>
          </div>
          <button className='btn btn--primary' ><Link to='/updateprofile'>Edit</Link></button>
          <div className="user" >
          <div className="userName">
            {lastName && <span>{lastName}</span>}
            {firstName && <span>{firstName}</span>}
          </div>
          <div className="moreInfo">
            {email && <span> <b>Email</b>: {email}</span>}
          </div>
          <div className="moreInfo">
            {department && <span> <b>Department</b>: {department}</span>}
          </div>
          <div className="moreInfo">
            {phone && <span> <b>Phone</b>: {phone}</span>}
          </div>
          <div className="moreInfo">
            {address && <span><b>Address</b>: {address}</span>}
          </div>
          <div className="moreInfo">
            {gender && <span><b>Gender</b>: {gender}</span>}
          </div>
          <div className="moreInfo">
            {role && <span><b>Role</b>: {role}</span>}
          </div>
          <div className="moreInfo">
            {currency && <span><b>Currency</b>: {currency}</span>}
          </div>
          </div>
          {this.props.error && <span className='error'>{this.props.error}</span>}
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  user: state.userProfile.user,
  error: state.userProfile.viewProfileError
});
const mapDispatchToProps = (dispatch) => ({
  fetchProfile: () => dispatch(viewProfile()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
