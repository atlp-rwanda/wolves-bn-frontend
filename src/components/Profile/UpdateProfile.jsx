import React, { Component } from 'react';
import { connect } from 'react-redux';
import InputField from '../common/InputField';
import Button from '../common/Button';
import { viewProfile, updateProfile } from '../../redux/actions/profile/profile';
import profileim from '../../assets/images/profileim.svg';
import './Profile.scss';

export class UpdateProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {
        photo: '',
        firstName: '',
        lastName: '',
        email: '',
        phone: '',
        address: '',
        gender: '',
        currency: '',
        department: ''
      },
      res: '',
    };

    this.onChange = this.onChange.bind(this);
    this.onFileChange = this.onFileChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentDidMount() {
    console.log(this.props.fetchProfile());
  }

  componentWillReceiveProps(nextProps) {
    this.setState(nextProps.user);
  }

  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  onFileChange(e) {
    this.setState({ [e.target.name]: e.target.files[0] });
  }

  onSubmit(e) {
    e.preventDefault();
    const data = {
      photo: this.state.photo,
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      phone: this.state.phone,
      email: this.state.email,
      address: this.state.address,
      gender: this.state.gender,
      currency: this.state.currency,
      department: this.state.department
    };
    const { updatingUser } = this.props;
    updatingUser(data);
  }

  render() {
    const message = this.props.updateError.Message;
    const {
      firstName, lastName, address, email, phone, currency, department, gender,
    } = this.state;
    return (
        <div className='container'>
          <div className='row content'>
            <div className='col-md-6'>
            <h3> <b>Update your profile information </b></h3>
            {this.props.res && <h4>{this.props.res}</h4>}
            <form action='#' onSubmit={this.onSubmit}>
            <InputField
            type='text'
            handleChange={this.onChange}
            label='First name'
            className='form-control'
            name='firstName'
            placeholder={firstName}
            />
            {this.props.updateError && <h5>{ message.includes('firstName') ? message : ''}</h5>}
            <InputField
            type='text'
            handleChange={this.onChange}
            label='Last name'
            className='form-control'
            name='lastName'
            placeholder={lastName}
            />
            {this.props.updateError && <h5>{ message.includes('lastName') ? message : ''} </h5>}
            <InputField
            type='text'
            handleChange={this.onChange}
            label='Address'
            className='form-control'
            name='address'
            placeholder={address}
            />
            {this.props.updateError && <h5>{ message.includes('address') ? message : ''} </h5>}
            <InputField
            type='email'
            handleChange={this.onChange}
            label='Email'
            className='form-control'
            name='email'
            placeholder={email}
            />
            {this.props.updateError && <h5>{ message.includes('email') ? message : ''} </h5>}
            <InputField
            type='number'
            handleChange={this.onChange}
            label='phone'
            className='form-control'
            name='phone'
            placeholder={phone}
            />
            {this.props.updateError && <h5>{ message.includes('phone') ? message : ''} </h5>}
            <InputField
            type='text'
            handleChange={this.onChange}
            label='Gender'
            className='form-control'
            name='gender'
            placeholder={gender}
            />
            {this.props.updateError && <h5>{ message.includes('gender') ? message : ''} </h5>}
            <InputField
            type='text'
            handleChange={this.onChange}
            label='Currency'
            className='form-control'
            name='currency'
            placeholder={currency}
            />
            {this.props.updateError && <h5>{ message.includes('currency') ? message : ''} </h5>}
            <InputField
            type='text'
            handleChange={this.onChange}
            label='Department'
            className='form-control'
            name='department'
            placeholder={department}
            />
            {this.props.updateError && <h5>{ message.includes('department') ? message : ''} </h5>}
            <InputField
            type="file"
            label='Profile image' department
            handleChange={this.onFileChange}
            name="photo"
            className="form-control"
            accept=".png, .jpg, .jpeg"
            />
            {this.props.updateError && <h5>{ message.includes('photo') ? message : ''} </h5>}
            <Button content='Update' />
          </form>
          </div>
          <div className='col-md-6 mb-3'>
              <img src={profileim} alt='' className='img-fluid' />
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  user: state.userProfile.user,
  error: state.userProfile.viewProfileError,
  updateError: state.userProfile.updateProfileError,
  res: state.userProfile.res
});

function mapDispatchToProps(dispatch) {
  return {
    updatingUser: (data) => dispatch(updateProfile(data)),
    fetchProfile: () => dispatch(viewProfile()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(UpdateProfile);