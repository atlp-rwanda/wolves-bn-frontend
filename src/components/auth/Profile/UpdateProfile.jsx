import React, { Component } from 'react';
import { connect } from 'react-redux';
import InputField from '../../common/InputField';
import Button from '../../common/Button';
import { viewProfile, updateProfile } from '../../../redux/actions/profile/profile';
import profileim from '../../../assets/images/profileim.svg';
import './Profile.scss';

export class UpdateProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {
        photo: '',
        firstName: '',
        lastName: '',
        email: '',
        phone: '',
        address: '',
        gender: '',
        currency: '',
        department: ''
      },
      res: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onFileChange = this.onFileChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentDidMount() {
    console.log(this.props.fetchProfile());
  }

  componentWillReceiveProps(nextProps) {
    this.setState(nextProps.user);
    console.log('nextProps.res', nextProps.res);
    console.log('nextProps', nextProps);
    if (nextProps.res) {
      this.props.history.push('/profile');
    }
  }

  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  onFileChange(e) {
    this.setState({ [e.target.name]: e.target.files[0] });
  }

  onSubmit(e) {
    e.preventDefault();
    const data = {
      photo: this.state.photo,
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      phone: this.state.phone,
      email: this.state.email,
      address: this.state.address,
      gender: this.state.gender,
      currency: this.state.currency,
      department: this.state.department
    };
    const { updatingUser } = this.props;
    updatingUser(data);
  }

  render() {
    const message = this.props.updateError.Message;
    const {
      firstName, lastName, address, email, phone, currency, department, gender,
    } = this.state;
    return (
        <div className='container'>
          <div className='my content'>
            <div className='mmmm'>
            <h3> <b>Update your profile information </b></h3>
            {this.props.res && <h4>{this.props.res}</h4>}
            <form action='#' onSubmit={this.onSubmit}>
            <InputField
            type='text'
            onChange={this.onChange}
            label='First name'
            className='form-control'
            name='firstName'
            placeholder={firstName}
            />
            {this.props.updateError && <h5>{ message.includes('firstName') ? message : ''}</h5>}
            <InputField
            type='text'
            onChange={this.onChange}
            label='Last name'
            className='form-control'
            name='lastName'
            placeholder={lastName}
            />
            {this.props.updateError && <h5>{ message.includes('lastName') ? message : ''} </h5>}
            <InputField
            type='text'
            onChange={this.onChange}
            label='Address'
            className='form-control'
            name='address'
            placeholder={address}
            />
            {this.props.updateError && <h5>{ message.includes('address') ? message : ''} </h5>}
            <InputField
            type='email'
            onChange={this.onChange}
            label='Email'
            className='form-control'
            name='email'
            placeholder={email}
            />
            {this.props.updateError && <h5>{ message.includes('email') ? message : ''} </h5>}
            <InputField
            type='number'
            onChange={this.onChange}
            label='phone'
            className='form-control'
            name='phone'
            placeholder={phone}
            />
            {this.props.updateError && <h5>{ message.includes('phone') ? message : ''} </h5>}
            <InputField
            type='text'
            onChange={this.onChange}
            label='Gender'
            className='form-control'
            name='gender'
            placeholder={gender}
            />
            {this.props.updateError && <h5>{ message.includes('gender') ? message : ''} </h5>}
            <InputField
            type='text'
            onChange={this.onChange}
            label='Currency'
            className='form-control'
            name='currency'
            placeholder={currency}
            />
            {this.props.updateError && <h5>{ message.includes('currency') ? message : ''} </h5>}
            <InputField
            type='text'
            onChange={this.onChange}
            label='Department'
            className='form-control'
            name='department'
            placeholder={department}
            />
            {this.props.updateError && <h5>{ message.includes('department') ? message : ''} </h5>}
            <InputField
            type="file"
            label='Profile image'
            onChange={this.onFileChange}
            label='Profile image'
            handleChange={this.onFileChange}
            name="photo"
            className="form-control"
            accept=".png, .jpg, .jpeg"
            />
            {this.props.updateError && <h5>{ message.includes('photo') ? message : ''} </h5>}
            <Button content='Update' />
          </form>
          </div>
          <div className='image'>
              <img src={profileim} alt='' className='img-fluid' />
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  user: state.userProfile.user,
  error: state.userProfile.viewProfileError,
  updateError: state.userProfile.updateProfileError,
  res: state.userProfile.res
});
function mapDispatchToProps(dispatch) {
  return {
    updatingUser: (data) => dispatch(updateProfile(data)),
    fetchProfile: () => dispatch(viewProfile()),
  };
}
export default connect(mapStateToProps, mapDispatchToProps)(UpdateProfile);