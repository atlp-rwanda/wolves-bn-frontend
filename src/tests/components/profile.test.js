import React from 'react';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import promise from 'redux-promise-middleware';
import thunk from 'redux-thunk';
import Profile from '../../components/Profile';

const middlewares = [thunk, promise];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

describe('profile component', () => {
  store.dispatch = jest.fn();

  const component = renderer.create(
    <Router>
    <Provider store={store}>
      <Profile />
    </Provider>
    </Router>
  );
  it('should render with initial state', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});