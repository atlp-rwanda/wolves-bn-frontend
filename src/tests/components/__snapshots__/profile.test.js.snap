// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile component should render with initial state 1`] = `
<div>
  <h2>
     Hello 
  </h2>
</div>
`;
